### ISSUE TITLE
nn.Parameter Overrides Global Gradient Settings

### ISSUE THREAD

**Issue by Radagaisus** [pytorch/pytorch#40008]
## 🐛 Bug: nn.Parameter Overrides Global Gradient Settings

`nn.Parameter` always sets `requires_grad` to `True` by default, even if the gradient has been disabled globally or with a `with` statement. It should first consult with the global and context settings before deciding on a default value.

## To Reproduce

```python
# Disable automatic differentiation
torch.autograd.set_grad_enabled(False)
# Even enter a no gradient context
with torch.no_grad():
  # In your model, setup a parameter:
  self.l1w = nn.Parameter(torch.randn(num_inputs, 20))
# l1w requires a gradient
assert self.l1w.requires_grad == True
```

## Expected behavior

`nn.Parameter` should first consult with the context and global settings before deciding on the default `requires_grad` behaviour.

cc @ezyang @SsnL @albanD @zou3519 @gqchen @mruberry

**Comment by ag14774** [comment]
This is not the use case for `torch.no_grad()`. If are planning to never record an operation on that tensor then just don't set it as a `nn.Parameter`. Or just call `.required_grad_(False)`. If you want to temporarily disable gradient computation then call `torch.no_grad()` *during* the computation/operation that involves `l1w` not during the creation of the parameter

**Comment by albanD** [comment]
Hi,

Yes this is expected behavior: grad mode only changes the behavior of operations. Not the creation of Tensors.
You can pass `requires_grad=False` to the `Parameter()` constructor if you want to set that flag on creation.
